libyul/backends/evm/EVMDialect.cpp

```diff
@@ -182,6 +182,7 @@ BuiltinFunctionForEVM const* EVMDialect::builtin(YulString _name) const
 EVMDialect const& EVMDialect::looseAssemblyForEVM(langutil::EVMVersion _version)
 {
 	static map<langutil::EVMVersion, unique_ptr<EVMDialect const>> dialects;
+	YulStringRepository::resetCallbacks().emplace_back([&] { dialects.clear(); });
```

Comment:
Where is the `YulString` reference in `dialects`? Or is YulString's `reset()` expected to reset non YulString references as well?

Reply:
It is inside `m_functions`

Reply:
okay, thanks!